File: test5/distinct_lines.py
#!/usr/bin/python3
import sys
import re
linecount = 0
uniq = 0
diction = {}

for line in sys.stdin:
	line = re.sub('\s+','',line )
	line = str.lower(line)
	linecount+=1
	if not line in diction:
		diction[line] = 1
		uniq +=1
	if uniq >= int(sys.argv[1]):
		print (uniq, "distinct lines seen after", linecount, "lines read.")
		sys.exit() 

print("End of input reached after", linecount, "lines read - ", sys.argv[1], "different lines not seen.")

File: test3/distinct_lines.pl
#!/usr/bin/perl -w
use strict;
my %hash  = ();
my $countUnique = 0;
my $countLines = 0;
while (my $line  = <STDIN>){
	$countLines++;
	$line =~ s/\s//g;
	$line = lc($line);
	if (!exists $hash{$line}){
		$countUnique++;
	} 
	

	$hash{$line} = 1;
	if($countUnique == $ARGV[0]){
		print"$countUnique distinct lines seen after $countLines lines read.\n"; 
		exit 0;	
	}

}
print"End of input reached after $countLines lines read - $ARGV[0] different lines not seen.\n";	
